Source/Math/NcclComm.cpp

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright (c) 2016, NVIDIA CORPORATION. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
```

Comment:
I'm will be adding a work item on clean up here.

---
In reply to: [87059775](https://github.com/Microsoft/CNTK/pull/874#discussion_r87059775) [](ancestors = 87059775,87001650)